File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.cpp
#include <string>
#include <vector>

using namespace std;

long long solution(int n) {
    
    long long dp[2001];
    
    dp[0] =0;   dp[1]=1;    dp[2]=2;
    
    for(int i=3;i<2001;i++)
    {
        dp[i] = (dp[i-2]+dp[i-1])%1234567;
    }
    return dp[n];
}

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.py
def solution(n):
    dp = []
    # dp 문제
    # 점화식 dp[i] = dp[i-2] + dp[i-1]
    
    dp.append(0) 
    dp.append(1) 
    dp.append(2) 
    dp.append(3) 
    dp.append(5)
    
    for i in range(5,2001,1) :
        dp.append((dp[i-2] + dp[i-1])%1234567)
        
    return dp[n]